src/pymmcore_widgets/_objective_widget.py

```diff
@@ -0,0 +1,123 @@
+        *,
+        mmcore: Optional[CMMCorePlus] = None
+    ):
+        super().__init__(parent)
+        self._mmc = mmcore or get_core_singleton()
+        self._objective_device = objective_device or self._guess_objective_device()
+        self._combo = self._create_objective_combo(objective_device)
+
+        lbl = QLabel("Objectives:")
+        lbl.setSizePolicy(QSizePolicy.Policy.Maximum, QSizePolicy.Policy.Maximum)
+
+        self.setLayout(QHBoxLayout())
+        self.layout().setContentsMargins(0, 0, 0, 0)
+        self.layout().addWidget(lbl)
+        self.layout().addWidget(self._combo)
+
+        self._mmc.events.systemConfigurationLoaded.connect(self._on_sys_cfg_loaded)
+        self.destroyed.connect(self._disconnect_from_core)
+        self._on_sys_cfg_loaded()
+
+    def _disconnect_from_core(self) -> None:
+        self._mmc.events.systemConfigurationLoaded.disconnect(self._on_sys_cfg_loaded)
+
+    def _on_sys_cfg_loaded(self) -> None:
+        """When config is loaded, check objective still exists, recreate combo."""
+        loaded = self._mmc.getLoadedDevices()
+        if self._objective_device not in loaded:
+            self._objective_device = None
+        if len(loaded) > 1:
+            if not self._objective_device:
+                self._objective_device = self._guess_objective_device()
+            self._clear_previous_device_widget()
+            self._combo = self._create_objective_combo(self._objective_device)
+            self.layout().addWidget(self._combo)
+
+    def _guess_objective_device(self) -> Union[str, None]:
+        """Try to update the list of objective choices.
+
+        1. get a list of potential objective devices from pymmcore
+        2. if there is only one, use it, if there are >1, show a dialog box
+        """
+        candidates = self._mmc.guessObjectiveDevices()
+        if len(candidates) == 1:
+            return candidates[0]
+        elif candidates:
+            dialog = ComboMessageBox(candidates, "Select Objective Device:", self)
+            if dialog.exec_() == dialog.DialogCode.Accepted:
+                return dialog._currentText()
+        return None
+
+    def _clear_previous_device_widget(self) -> None:
+        self._combo.setParent(None)
+        self._combo.deleteLater()
+
+    def _create_objective_combo(
+        self, device_label: Union[str, None]
+    ) -> Union[StateDeviceWidget, QComboBox]:
+        if device_label:
+            combo = _ObjectiveStateWidget(device_label, mmcore=self._mmc)
+            combo.setMinimumWidth(285)
+            combo._combo.currentIndexChanged.connect(self._on_obj_changed)
+        else:
+            combo = QComboBox()
+            combo.setEnabled(False)
+        return combo
+
+    def _on_obj_changed(self) -> None:
+        self._mmc.events.pixelSizeChanged.emit(self._mmc.getPixelSizeUm() or 0.0)
+
+
+class _ObjectiveStateWidget(StateDeviceWidget):
+    """
+    Subclass of StateDeviceWidget.
```

Comment:
put summary comment on the first line:

```suggestion
    """Subclass of StateDeviceWidget.
```

Also ... try to avoid comments that literally say what the code says (the line above says that this is just a subclass of state device widget).  Either skip the comment entirely for a private class, or give a one-line explanation for what it _does_ (not what it is)